postinstall.js

```diff
@@ -0,0 +1,49 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+/* eslint-disable no-console */
+
+const fsPromises = require('fs/promises');
+
+const syntheticsFormatter = 'node_modules/@elastic/synthetics/dist/formatter/javascript.js';
+const requirePWCore = 'require("playwright-core/lib/server/recorder/javascript");';
+const requirePW = 'require("playwright/lib/server/recorder/javascript");';
+
+async function replace() {
+  try {
+    const formatterText = await fsPromises.readFile(syntheticsFormatter, { encoding: 'utf-8' });
+    if (formatterText.indexOf(requirePWCore) !== -1) {
+      const updatedFormatterText = formatterText.replace(requirePWCore, requirePW);
+      await fsPromises.writeFile(syntheticsFormatter, updatedFormatterText);
+      console.log(
+        '`@elastic/synthetics` formatter reference changed from `playwright-core` to `playwright`'
+      );
+    } else {
+      console.log('Skipping postinstall change of `@elastic/synthetics` formatter.');
+    }
+  } catch (e) {
+    console.error('Could not update Synthetics formatter', e);
+  }
+}
+
+replace();
```

Comment:
interesting can you call an async function in top level during postinstall?

Reply:
I was worried about this as well for a moment but it seems to work without problems.